src/scss/parse.js

```diff
@@ -3025,14 +3028,18 @@ function checkPseudoClass3(i) {
   if (l = checkSC(i)) i += l;
 
   if (i >= tokensLength) return 0;
-  if (tokens[i].value === '+' ||
-      tokens[i].value === '-') i++;
+
+  if (tokens[i].type === TokenType.PlusSign ||
+      tokens[i].type === TokenType.HyphenMinus) i++;
   else return 0;
 
   if (l = checkSC(i)) i += l;
 
+  if (l = checkInterpolation(i)) i += l;
+
   if (tokens[i].type === TokenType.DecimalNumber) i++;
-  else return 0;
```

Comment:
Here you make both interpolation and number optional. That means that following expressions will become valid:
```
:nth-child(2n -)
:nth-child(2n - #{$x}3)
:nth-child(2n - 3#{$x})
:nth-child(2n - #{$x}#{$y})
```
Is it by intention?